File: src/VisualStudio/CSharp/Impl/Options/AutomationObject/AutomationObject.DocumentationComment.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using Microsoft.CodeAnalysis.DocumentationComments;

namespace Microsoft.VisualStudio.LanguageServices.CSharp.Options;

public partial class AutomationObject
{
    public int AutoComment
    {
        get { return GetBooleanOption(DocumentationCommentOptionsStorage.AutoXmlDocCommentGeneration); }
        set { SetBooleanOption(DocumentationCommentOptionsStorage.AutoXmlDocCommentGeneration, value); }
    }

    public int GenerateSummaryTagOnSingleLine
    {
        get { return GetBooleanOption(DocumentationCommentOptionsStorage.GenerateSummaryTagOnSingleLine); }
        set { SetBooleanOption(DocumentationCommentOptionsStorage.GenerateSummaryTagOnSingleLine, value); }
    }

    public int GenerateOnlySummaryTag
    {
        get { return GetBooleanOption(DocumentationCommentOptionsStorage.GenerateOnlySummaryTag); }
        set { SetBooleanOption(DocumentationCommentOptionsStorage.GenerateOnlySummaryTag, value); }
    }
}

File: src/VisualStudio/VisualBasic/Impl/Options/AutomationObject/AutomationObject.DocumentationComment.vb
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
' See the LICENSE file in the project root for more information.

Imports Microsoft.CodeAnalysis.DocumentationComments

Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.Options
    Partial Public Class AutomationObject
        Public Property AutoComment As Boolean
            Get
                Return GetBooleanOption(DocumentationCommentOptionsStorage.AutoXmlDocCommentGeneration)
            End Get
            Set(value As Boolean)
                SetBooleanOption(DocumentationCommentOptionsStorage.AutoXmlDocCommentGeneration, value)
            End Set
        End Property

        Public Property GenerateSummaryTagOnSingleLine As Boolean
            Get
                Return GetBooleanOption(DocumentationCommentOptionsStorage.GenerateSummaryTagOnSingleLine)
            End Get
            Set(value As Boolean)
                SetBooleanOption(DocumentationCommentOptionsStorage.GenerateSummaryTagOnSingleLine, value)
            End Set
        End Property

        Public Property GenerateOnlySummaryTag As Boolean
            Get
                Return GetBooleanOption(DocumentationCommentOptionsStorage.GenerateOnlySummaryTag)
            End Get
            Set(value As Boolean)
                SetBooleanOption(DocumentationCommentOptionsStorage.GenerateOnlySummaryTag, value)
            End Set
        End Property
    End Class
End Namespace
